feat(example): add examples of effects not based on the Actions stream

adrianfaciu opened the pull request:
As suggested, added two examples of effects not based on Actions stream:
- listen for router navigation events and update the page title
- log out the user after a specified period of inactivity

Closes #1830

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Closes #1830

## What is the new behavior?

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

## Other information

timdeschryver commented:
Another idea that pops to my mind would be to use google analytics. 
We could NgRx-ify Stephen's post https://fluin.io/blog/google-analytics-with-tag-manager-and-angular

timdeschryver commented:
An example that still uses the actions but are not dispatching an action back to the store:
Showing a [snackbar](https://material.angular.io/components/snack-bar/overview) when a book has been added/removed.

brandonroberts commented on projects/example-app/src/app/core/actions/user.actions.ts:
```diff
@@ -0,0 +1,3 @@
+import { createAction } from '@ngrx/store';
+
+export const idle = createAction('[User] Idle');
```
```suggestion
export const idleTimeout = createAction('[User] Idle Timeout');
```

brandonroberts commented on projects/example-app/src/app/core/effects/router.effects.ts:
```diff
@@ -0,0 +1,32 @@
+    () =>
+      this.router.events.pipe(
+        filter(event => event instanceof NavigationEnd),
+        map(() => this.activatedRoute),
```
Combine the 2 maps into 1

brandonroberts commented on projects/example-app/src/app/core/effects/router.effects.ts:
```diff
@@ -0,0 +1,32 @@
+          return route;
+        }),
+        mergeMap(route => route.data),
+        map(data => data['title']),
```
```suggestion
        map(data => `Book Collection - ${data['title'])}`,
```

brandonroberts commented on projects/example-app/src/app/core/effects/router.effects.ts:
```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import { createEffect } from '@ngrx/effects';
+import { Router, NavigationEnd, ActivatedRoute } from '@angular/router';
```
Move all `@angular` imports to the top of the file, following by `rxjs` and `@ngrx`

brandonroberts commented on projects/example-app/src/app/core/effects/user.effects.ts:
```diff
@@ -0,0 +1,21 @@
+
+  idle$ = createEffect(() =>
+    merge(this.clicks$, this.keys$, this.mouse$).pipe(
+      switchMapTo(timer(60 * 1000)),
```
```suggestion
      switchMapTo(timer(5 * 60 * 1000)), // 5 minute inactivity timeout
```

adrianfaciu commented:
Updated with all the requested changes and added tests for user and router effects.
If we need more examples, I can add some effects for the snackbar and look into google analytics.

alex-okrushko commented on projects/example-app/src/app/core/effects/router.effects.ts:
```diff
@@ -0,0 +1,34 @@
+export class RouterEffects {
+  updateTitle$ = createEffect(
+    () =>
+      this.router.events.pipe(
```
We have store-router package installed in the example app. Wouldn't that be better to listen to those changes?
@brandonroberts

adrianfaciu replied:
I thought about that when I was writing the code :) The idea was to show how one would create effects not based on the Actions stream, so I went for this instead.

brandonroberts replied:
I think these are good for what we asked for
